rct229/ruleset_functions/baseline_systems/is_baseline_system_5.py

```diff
@@ -0,0 +1,117 @@
+    does_each_zone_have_only_one_terminal,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_cooling_type_DX import (
+    is_hvac_sys_cooling_type_dx,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_fan_sys_VSD import (
+    is_hvac_sys_fan_sys_vsd,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_preheat_fluid_loop_attached_to_boiler import (
+    is_hvac_sys_preheat_fluid_loop_attached_to_boiler,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_preheat_fluid_loop_purchased_heating import (
+    is_hvac_sys_preheat_fluid_loop_purchased_heating,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_preheating_type_fluid_loop import (
+    is_hvac_sys_preheating_type_fluid_loop,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_system_util import (
+    HVAC_SYS,
+    find_exactly_one_hvac_system,
+)
+
+HEATING_SYSTEM = schema_enums["HeatingSystemOptions"]
+COOLING_SYSTEM = schema_enums["CoolingSystemOptions"]
+
+
+def is_baseline_system_5(rmi_b, hvac_b_id, terminal_unit_id_list, zone_id_list):
+    """
+     Get either Sys-5, Sys-5b, or Not_Sys_5 string output which indicates whether the HVAC system is ASHRAE 90.1 2019 Appendix G system 5 (Package VAV with Reheat) or system 5b (system 5 with purchased heating).
+
+     Parameters
+     ----------
+     rmi_b json
+         To evaluate if the hvac system is modeled as either Sys-11.1, Sys-11.1a, Sys-11b, Sys-11c, or Not_Sys_11.1 in the B_RMI.
+     hvac_b_id list
+         The id of the hvac system to evaluate.
+     terminal_unit_id_list
+         List list of terminal unit IDs associated with the HVAC system to be evaluated. These are sent to this function from the master get_baseline_system_types function.
+     zone_id_list list
+         list of zone IDs associated with the HVAC system to be evaluated. These are sent to this function from the master get_baseline_system_types function.
+
+    Returns
+     -------
+     The function returns either Sys-5, Sys-5b, or Not_Sys_5 string output which indicates whether the HVAC system is ASHRAE 90.1 2019 Appendix G system 5 (Package VAV with Reheat) or system 5b (system 5 with purchased heating).
+    """
+
+    is_baseline_system_5 = HVAC_SYS.UNMATCHED
+
+    hvac_b = find_exactly_one_hvac_system(rmi_b, hvac_b_id)
+    # check if the hvac system has the required sub systems for system type 11.1
+    has_required_heating_sys = (
+        hvac_b.get("heating_system") is None
+        or hvac_b["heating_system"].get("heating_system_type") is None
+        or hvac_b["heating_system"]["heating_system_type"] == HEATING_SYSTEM.NONE
+    )
+
+    has_required_preheat_sys = (
+        hvac_b.get("preheat_system")
+        is not None  # if preheat system exists, the number must be 1.
```

Comment:
Need to update the RDS for preheat system

Reply:
Great point. Fixed.